File: irssi/scripts/notesay.py
# -*- coding: utf-8 -*-
import random

NOTES = u'♩♪♫♬'
MODIFIERS = u'♭♮♯'

def picknote():
    ret = random.choice(NOTES)
    if not random.randrange(4):
        ret = random.choice(MODIFIERS) + ret
    return ret

def notesay(s):
    return u'%s %s %s' % (picknote(), s, picknote())

def irssi_notesay(data, server, witem):
    data = data.decode('utf-8')
    witem.command('say %s' % (notesay(data).encode('utf-8'),))

if __name__ == '__main__':
    import sys
    print notesay(' '.join(sys.argv[1:]).decode('utf-8')).encode('utf-8')
else:
    import irssi
    irssi.command_bind('notesay', irssi_notesay)

File: irssi/scripts/notesay.pl
use warnings;
use strict;
use utf8;

my @NOTES = qw/♩ ♪ ♫ ♬/;
my @MODIFIERS = qw/♭ ♮ ♯/;

sub random_choice {
    return @_[int rand scalar @_];
}

sub picknote {
    my $note = random_choice(@NOTES);
    if (int rand 4 == 0) {
        $note = random_choice(@MODIFIERS) . $note;
    }
    return $note;
}

sub notesay {
    my ($lyric) = @_;
    return picknote() . " $lyric " . picknote();
}

sub cmd_notesay {
    my ($data, $server, $witem) = @_;
    $data = notesay($data);
    $witem->command("say $data");
}

eval { Irssi::Core::is_static() };
if ($@) {
    use 5.010;
    binmode STDOUT, ":encoding(UTF-8)";
    say notesay($ARGV[0]);
} else {
    require Irssi;
    Irssi->import;
    Irssi::command_bind('notesay', 'cmd_notesay');
}
